### PR TITLE
Bugfix in load balancing routine

### LINKED ISSUE #2554: Load balance high residual and seg fault with EB
When trying to use load balancing with the input file from https://github.com/ECP-WarpX/WarpX/pull/2538, with the changes from https://github.com/ECP-WarpX/WarpX/pull/2401, and adding return statements to the beginning of `WarpX::ComputeEdgeLengths`, `WarpX::ComputeFaceAreas`, `WarpX::ScaleEdges`, and `WarpX::ScaleAreas`, we get a very high residual (3e289) on step 4 (load_balance_intervals = 5), causing the run to fail and exit. If we set a high absolute tolerance for the MLMG solver (3e290) to bypass this high residual, the program will segfault at the beginning of step 5 with backtraces from some of the processes.

The input file we used is [inputs2d.txt](https://github.com/ECP-WarpX/WarpX/files/7529917/inputs2d.txt).

This is one backtrace file we got using a debug build, when running with the conditions above to reach the segfault on step 5. This simulation was run with 8 mpi cores and 1 omp thread. [Backtrace.0.0.txt](https://github.com/ECP-WarpX/WarpX/files/7529925/Backtrace.0.0.txt)

### PR DESCRIPTION
**Pull request by roelof-groenewald**
Add remake of phi_fp during load balancing RemakeLevel.
Partial fix for #2554.

### CODE REVIEW

**Comment by PhilMiller** [comment]
Modification looks good to me. But wow, is that function screaming to have a helper defined!

File: Source/Parallelization/WarpXRegrid.cpp [hunk 1]
```diff
@@ -209,6 +209,14 @@ WarpX::RemakeLevel (int lev, Real /*time*/, const BoxArray& ba, const Distributi
             rho_fp[lev] = std::move(pmf);
         }
 
+        if (phi_fp[lev] != nullptr) {
+            const int nc = phi_fp[lev]->nComp();
+            const IntVect& ng = phi_fp[lev]->nGrowVect();
+            auto pmf = std::make_unique<MultiFab>(phi_fp[lev]->boxArray(),
+                                                              dm, nc, ng);
+            phi_fp[lev] = std::move(pmf);
+        }
```
Comment (RemiLehe) [inline Source/Parallelization/WarpXRegrid.cpp:218]: Could you also add `phi_cp` further below?

**Comment by ax3l** [comment]
X-Ref: Follow-up to #1531

**Review by ax3l — APPROVED** [review]
Thanks a lot!

Will assign an extra set of eyes.

**Review by dpgrote — APPROVED** [review]
Good catch. This looks good to me

**Review by RemiLehe — DISMISSED** [review]
Thanks for this PR! Could you do the same change for `phi_cp` (This would be used for mesh refinement.)

File: Source/Parallelization/WarpXRegrid.cpp (hunk 1 above)
Comment (roelof-groenewald) [reply Source/Parallelization/WarpXRegrid.cpp:218]: No problem.

Comment (roelof-groenewald) [reply Source/Parallelization/WarpXRegrid.cpp:218]: Actually it doesn't look like `phi_cp` is defined anywhere.

Comment (RemiLehe) [reply Source/Parallelization/WarpXRegrid.cpp:218]: Ah, yes, that's right. I got confused.

**Pull request merged**